rllib/policy/policy.py

```diff
@@ -58,6 +60,22 @@ class you pass into RolloutWorker will be constructed with
         self.observation_space = observation_space
         self.action_space = action_space
         self.config = config
+        # The global timestep, broadcast down from time to time from the
+        # driver.
+        self.global_timestep = 0
+
+        # Create the Exploration object to use for this Policy.
+        self.exploration = from_config(
+            Exploration,
+            config.get("exploration"),
+            action_space=self.action_space,
+            framework="torch" if self.config.get("use_pytorch") else "tf",
+            worker_info=self.config.get("worker_info"))
+        self.last_exploration_info = []
```

Comment:
Should this be dict type?

Reply:
Maybe better, yup.

Reply:
Or return anything that's "summarizable". I changed it to return the plain epsilon float value for simplicity. Other Exploration classes can return something else, a dict, whatever. This is really just for reporting purposes, it's not the state of the Exploration.